src/Component.js

```diff
@@ -46,6 +46,9 @@ export const DebounceInput = React.createClass({
 
 
   componentWillReceiveProps({value, debounceTimeout}) {
+    if (this.isDebouncing) {
```

Comment:
Yea, this is the scariest part.... Totally understand. No rush.

If you want, I can hide this behind an optional prop to make this a little safer backwards-compatibility-wise.